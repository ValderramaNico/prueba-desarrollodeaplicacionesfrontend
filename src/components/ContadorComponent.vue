<template>
  <h2>{{ tituloApp }}</h2>
  <p class="counter">{{ contador }}</p>

  <div>
    <button class="btn primary" v-on:click="incrementar">+1</button>
    <button class="btn warning" v-on:click="disminuir">-1</button>
  </div>
</template>

<script>
export default {
  name: "ContadorComponent",
  props: {
    titulo: String,
    inicio: {
      type: Number,
      default: 0,
    },
  },

  data() {
    return {
      contador: this.inicio,
    };
  },
  methods: {
    incrementar() {
      this.contador++;
    },

    disminuir() {
      this.contador--;
    },
  },
  computed: {
    tituloApp() {
      return this.titulo || "Contador";
    },
  },
};
</script>

<style>
.btn {
  border-radius: 1rem;
  margin: 0 5px;
  width: 100px;
  padding: 5px;
  font-size: 1rem;
}
.primary {
  background-color: rgb(15, 15, 133);
  color: rgb(255, 255, 255);
}
.counter {
  font-size: 36px;
}

.warning {
  background-color: rgb(163, 15, 15);
  color: rgb(255, 255, 255);
}
</style>
