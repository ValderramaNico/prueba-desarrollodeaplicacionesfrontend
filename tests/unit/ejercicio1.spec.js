import { shallowMount } from "@vue/test-utils";
import ContadorComponent from "@/components/ContadorComponent.vue";


describe("Test1 - ContadorComponent", () => {
  it("Test1 - Se recibe un valor inicial de 0 en el contador.", () => {
    const wrapper = shallowMount(ContadorComponent, { //Se monta el componente y se le entregan las props para testear, en este caso, un valor en 0.
      props: {
        inicio: 0
      },
    });
    expect(wrapper.find("p").text()).toBe("0"); //Se valida que el valor en el contador con la etiqueta '<p>' sea de 0, después de settearlo en 'props' a '0'
    expect(wrapper.html()).toMatchSnapshot(); // Capturamos un snapshot del HTML del componente  
  });
});

describe("Test2 - ContadorComponent", () => {
  it("Test 2 - Debe incrementar el contador en 1 cuando se hace click en el botón +1", async () => {
    const wrapper = shallowMount(ContadorComponent, { //Se monta el componente y se le entregan las props para testear el boton, en este caso, un valor en 0
      props: {
        inicio: 0
      },
    });

    await wrapper.find("button.primary").trigger("click"); //Se hace un '.trigger' al botón de 'class="button.primary" (se usa la clase para identificarlo)'
    expect(wrapper.find("p").text()).toBe("1"); //Se verifica que el contador haya sumado desde el valor inicial 0 y tenga como valor '1'
    expect(wrapper.html()).toMatchSnapshot(); // Capturamos un snapshot del HTML del componente
  });

  it("Test2 - Debe decrementar el contador en 1 cuando se hace click en el botón -1", async () => {
    const wrapper = shallowMount(ContadorComponent, {
      props: {
        inicio: 0
      },
    });

    await wrapper.find("button.warning").trigger("click"); //Se hace un '.trigger' al botón de 'class="button.primary" (se usa la clase para identificarlo)'
    expect(wrapper.find("p").text()).toBe("-1"); //Se verifica que el contador haya restado desde el valor inicial 0 y tenga como valor '-1'
    expect(wrapper.html()).toMatchSnapshot(); // Capturamos un snapshot del HTML del componente
  });
});
