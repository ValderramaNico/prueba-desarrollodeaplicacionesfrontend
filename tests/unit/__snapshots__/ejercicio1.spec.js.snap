// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test1 - ContadorComponent Test1 - Se recibe un valor inicial de 0 en el contador. 1`] = `
<h2>Contador</h2>
<p class="counter">0</p>
<div><button class="btn primary">+1</button><button class="btn warning">-1</button></div>
`;

exports[`Test2 - ContadorComponent Test 2 - Debe incrementar el contador en 1 cuando se hace click en el botón +1 1`] = `
<h2>Contador</h2>
<p class="counter">1</p>
<div><button class="btn primary">+1</button><button class="btn warning">-1</button></div>
`;

exports[`Test2 - ContadorComponent Test2 - Debe decrementar el contador en 1 cuando se hace click en el botón -1 1`] = `
<h2>Contador</h2>
<p class="counter">-1</p>
<div><button class="btn primary">+1</button><button class="btn warning">-1</button></div>
`;
